### PR TITLE
Adds KaOS support

### PR DESCRIPTION
**Pull request by ShalokShalom**
See this: https://github.com/commercialhaskell/stack/issues/5452

Is this something I am supposed to mention in the change.log?

Then I do it

### CODE REVIEW

File: etc/scripts/get-stack.sh [hunk 1]
```diff
@@ -328,6 +328,31 @@ do_alpine_install() {
...
+        *) printf "Please answer with y or n. ";;
+      esac
+    done
+
+    while true; do
+      echo
+      printf  "To be detectable by IDEs and other applications is it recommended to symlink \e[34mGHC\e[0m to the PATH.\nDo you like to do that?" 
+      read -p " y or n: " yn
+    case $yn in
+        [Yy]*)  sudo ln -s /home/$USER/.stack/programs/x86_64-linux/ghc-tinfo6-8.8.4/bin/ghc /usr/bin; break;;
```
Comment (ShalokShalom) [inline etc/scripts/get-stack.sh:350]: This line can only work when we soft code the version

**Review by ShalokShalom — COMMENTED** [review]
I did a mistake and hardcoded that one, how can we detect the version of ghc-tinfo6 easily?

File: etc/scripts/get-stack.sh (hunk 1 above)
Comment (hasufell) [reply etc/scripts/get-stack.sh:350]: I don't think install scripts should ever do this. `/usr/bin` is controlled by your distros package manager.

Also note that there is a `sudocmd` function:

https://github.com/commercialhaskell/stack/blob/708b516a8d34cc42b6c533d161b18b31f913c3d6/etc/scripts/get-stack.sh#L138

Comment (ShalokShalom) [reply etc/scripts/get-stack.sh:350]: So you suggest a different dir for the PATH?
And using sudocmd instead sudo?

Comment (hasufell) [reply etc/scripts/get-stack.sh:350]: My opinion is to not do anything here. Global GHC installations is probably not in the scope the install script. IDEs like HLS work with stack just fine without exposing a global GHC.

Comment (ShalokShalom) [reply etc/scripts/get-stack.sh:350]: its exactly what this script does for Debian and Co, no?

Comment (ShalokShalom) [reply etc/scripts/get-stack.sh:350]: And vscode does not

Comment (hasufell) [reply etc/scripts/get-stack.sh:350]: > its exactly what this script does for Debian and Co, no?

I don't think so.

**Comment by mpilgrem** [comment]
I would like to resolve this pull request, one way or another. I am a Windows user, so forgive me if I take things step-by-step and ask questions that may seem obvious to a Linux user.

I understand that KaOS is a Linux distribution that is x86-64-specific, and `kcp` is a KaOS package manager. I also understand from the above that KaOS can use the generic Linux GHC binary, but needs `kcp` package `libtinfo`(https://kaos-community-packages.github.io/items/libtinfo.html) as a pre-requisite. Is that correct? Does KaOS not come with something that other Linux distributions provide as standard?

Other parts of the script (for other Linux distribtions) have things like this:

~~~sh
do_fedora_install() {
  install_dependencies() {
    dnf_install_pkgs perl make automake gcc gmp-devel libffi zlib-devel xz tar git gnupg
  }

  if is_x86_64 ; then
    install_dependencies "$1"
    print_bindist_notice
    install_x86_64_linux_binary
  else
    die "Sorry, currently only 64-bit (x86_64) Linux binary is available."
  fi
}

# Install packages using dnf
dnf_install_pkgs() {
  if ! sudocmd "install required system dependencies" dnf install -y ${QUIET:+-q} "$@"; then
    die "\nInstalling dnf packages failed.  Please run 'dnf check-update' and try again."
  fi
}
~~~

Should KaOS/kcp follow the same 'pattern' for its `libtinfo` need? What is it about `libtinfo` that a warning is required before it is installed?

Is `libtinfo` the only dependency that is required if the distribution is KaOS? Distributions such as Ubuntu and Fedora seem to have a list of things that are needed?

I wondered why KaOS is introduced via `do_sloppy_install()` and not as its own `case "$DISTRO"` in `do_distro()`, if it is now being catered for expressly?

The addition of a link in ` /usr/bin` to a particular version of GHC in Stack's 'sandbox' looks wrong to me. If a KaOS user also wanted a specific version of GHC on the PATH (as well as the Stack-supplied ones), would they not just install it?

**Comment by ShalokShalom** [comment]
Wonderful 😄 

Yes, you are basically correct about everything - the only thing I like to mention, is that libtinfo is build by the community and kcp is also a package manager for the community repo. 

I only asked, if said dependency is fine to be added, since some people are quite sensitive about that. 

I personally think its perfectly fine. 👍🏼 

I love to introduce KaOS on the same level and with the same syntax as other distros, I think I was just to shy and hesitant to actually change something of the existing code. 

You are also right about hard coding a specific version. 

The only thing that is important to me, is to have an option to let the user choose to automatically add the compiler to the PATH. 

Preferably, by scanning the sub directories, or simply by choosing the most recent version. 

This way, people can just start coding. 🥳

**Comment by ShalokShalom** [comment]
Sorry, I am not on KaOS anymore.

**Comment by mpilgrem** [comment]
I asked for testers on the [Haskell Community](https://discourse.haskell.org/t/kaos-linux-distro-and-stacks-installation-script/4953) but had no response. I am going to close this pull request. It can be reopened if another KaOS user and Haskeller emerges.

**Comment by ShalokShalom** [comment]
I just have no time at the moment. I tried to launch it in Virtualbox, but failed. 

Will try if I find the time

**Pull request closed without merging**